contracts/usul/BaseTokenVoting.sol

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/draft-EIP712Upgradeable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "./BaseStrategy.sol";
+
+/// @title An abstract contract used as a base for ERC-20 token voting strategies
+abstract contract BaseTokenVoting is BaseStrategy {
+    enum VoteType {
+        Against,
+        For,
+        Abstain
+    }
+
+    struct ProposalVoting {
+        uint256 yesVotes; // The total number of YES votes for this proposal
+        uint256 noVotes; // The total number of NO votes for this proposal
+        uint256 abstainVotes; // The total number of ABSTAIN votes for this proposal
+        uint256 deadline; // The timestamp voting ends for this proposal
+        uint256 startBlock; // The block the proposal voting starts
+        mapping(address => bool) hasVoted;
+    }
+
+    uint256 public votingPeriod; // the length of time voting is valid for a proposal
+    uint256 public timelockPeriod;
+    string public name;
+
+    mapping(uint256 => ProposalVoting) internal proposals;
+
+    event TimelockPeriodUpdated(uint256 newTimeLockPeriod);
```

Comment:
super nitpicky but we should be consistent with making Timelock a single word, so this should be `newTimelockPeriod` (lowercase L)

Reply:
Fixed this